### DISCUSSION TITLE
Ext configuration of Mindconnect Agent node

### DISCUSSION THREAD

**Discussion by AlxLifanov** [mindsphere/node-red-contrib-mindconnect#145] [Ideas]
Greetings!

We need to make IOT2050 + Nodered + Mindconnect node to be onboarded by unqualified personnel on site.
Idea: we send a text file to the customer, which contains needed SHARED_SECRET. This file needs to be written to USB key to root folder. Nodered periodically checks for removable drive presence. If drive and file are present - credentials should be copied to Agent node.

Problem: SHARED_SECRET now is a constant and can be copied only via web interface. Is it possible to pass agent configuration programmaticaly from external source?

**Comment by sn0wcat** [comment]
@AlxLifanov  you can send the complete node configuration (including the secrets) as an input message to the node. This means that you can have a flow directly in node-red which can watch a folder and forward the configuration to the node once it was copied. After the configuration arrives the node will reconfigure itself and restart the flow. Below is the example how to send a managed configuration to the node.

```javascript
msg.payload ={
    "name": "Managed Configuration",
    "configtype": "SHARED_SECRET",
    "agentconfig": {
        "content": {
            "baseUrl": "https://southgate.eu1.mindsphere.io",
            "iat": "eyJ...Mkw",
            "clientCredentialProfile": ["SHARED_SECRET"],
            "clientId": "d...",
            "tenant": "c..."
        },
        "expiration": "2018-11-15T17:31:35.000Z"
    },
    "privatekey": "",
    "model": "",
    "validate": true,
    "validateevent": true,
    "chunk": true,
    "disablekeepalive": false,
    "retry": "5",
    "parallel": "7",
    "asyncduration": "10",
    "emitcontrol": true,
    "datalakeonly": true,
    "supressverbosity": true
}


return msg;

**Reply by AlxLifanov** [reply]
Cool, right thing. Missed that in documentation (if it is).

**Reply by sn0wcat** [reply]
it is not in documentation (yet) ...i was playing with the idea to implement the remote configuration of the nodes, so i implemented already the basic infrastructure for it a while ago but I had to postpone it due to lack of time...
